hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/command/CreateHoodieTableCommand.scala

```diff
@@ -179,7 +174,9 @@ object CreateHoodieTableCommand {
     val client = HiveClientUtils.newClientForMetadata(sparkSession.sparkContext.conf,
       sparkSession.sessionState.newHadoopConf())
     // create hive table.
-    client.createTable(tableWithDataSourceProps, ignoreIfExists)
+    if (!client.tableExists(table.identifier.database.get, table.identifier.table)) {
+      client.createTable(tableWithDataSourceProps, ignoreIfExists)
+    }
```

Comment:
in this case, the `ignoreIfExists` variable is useless; it should always be like `client.createTable(tableWithDataSourceProps, true)` ? then we don't need the if condition and we can remove `ignoreIfExists`